src/QueryReflection/RuntimeConfiguration.php

```diff
@@ -59,6 +59,11 @@ final class RuntimeConfiguration
      */
     private $utilizeSqlAst = false;
 
+    /**
+     * @var bool
+     */
+    private $doctrineKeyValueIntegerRangeChecks = false;
```

Comment:
I think we dhould have a more generic name and and use the same switch for parameter validation in other rules

```suggestion
    /** @var 'lax'|'strict' */
    private $parameterTypeValidation ='lax';
```

Reply:
I'm certainly happy to change the option name. Could I ask, what did you envision using it for? My intention was for it to be specific to integer range types, since in my use case I don't think I'd want "lax" validation for any other type. Would you be okay with something like `strictIntegerRangeChecks = bool` (or `integerRangeChecks = 'lax'|'strict'`)?

Reply:
So you don't want strict/lax validation for e.g. numeric-string/non-falsy-string/non-empty-string/string ?

Reply:
> So you don't want strict/lax validation for e.g. numeric-string/non-falsy-string/non-empty-string/string ?

With the integer range checks, the issue is that the parameter types (`IntegerType`, in most of my cases) are wider than the database type (`IntegerRangeType`), and so a strict check fails for those parameters. Is the same scenario possible for strings? Looking at `MysqlTypeMapper`, it seems that string-like columns are all mapped to `StringType`, and so valid parameter types shouldn't be wider than that, even if we're being lax.

However, I could definitely be missing something here. Do you have an example in mind? I hope this doesn't come across as being difficult, I'm genuinely trying to learn so that I can become a better contributor to this repo.

Either way, I won't be able to follow-up until tomorrow, as it is midnight here. Thanks for your patience. :)

Reply:
Sounds good, I went with your original suggestion, then. If we end up needing more fine-grained control in the future, hopefully it won't be too much trouble to add extra options (or perhaps extra modes for this option).

Thanks for the kind words :)